spec.bs

```diff
@@ -221,6 +224,35 @@ The <dfn for="interest group">estimated size</dfn> of an [=interest group=] |ig|
     [=interest group ad/render url=].
   1. The [=string/length=] of |ad|'s [=interest group ad/metadata=] if the field is not null.
 
+<div algorithm="check interest group permissions">
+
+To <dfn>check interest group permissions</dfn> given a [=string=] <dfn
+for="fetch interest group permissions">|ownerOrigin|</dfn>, a [=string=] <dfn

```

Comment:
I don't think the parameter names should all be "defined" as dfns. You can just mention their types as you do and put their names in pipes like you do, without the dfn tag, like you do in the last algorithm of this PR.